Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -96,25 +130,47 @@ public final class RewardAddOnSelectionViewModel: RewardAddOnSelectionViewModelT
       selectedQuantities
     )
 
-    self.loadAddOnRewardsIntoDataSourceAndReloadTableView = rewardAddOnCardsViewData
+    let reloadRewardsIntoDataSource = rewardAddOnCardsViewData
       .withLatest(from: latestSelectedQuantities)
       .map(rewardsData)
 
+    self.loadAddOnRewardsIntoDataSourceAndReloadTableView = Signal.merge(
+      reloadRewardsIntoDataSource,
+      requestErrored.filter(isTrue).mapConst([.emptyState(.errorPullToRefresh)])
+    )
+
     self.loadAddOnRewardsIntoDataSource = rewardAddOnCardsViewData
       .takePairWhen(latestSelectedQuantities)
       .map(rewardsData)
 
     self.shippingLocationViewIsHidden = baseReward.map(\.shipping.enabled)
       .negate()
 
+    let dataSourceItems = Signal.merge(
+      self.loadAddOnRewardsIntoDataSourceAndReloadTableView,
+      self.loadAddOnRewardsIntoDataSource
+    )
+
+    let allRewards = dataSourceItems.map { items in
+      items.compactMap { item -> Reward? in item.rewardAddOnCardViewData?.reward }
+    }
+
+    let baseRewardAndAddOnRewards = Signal.combineLatest(
+      baseReward,
+      dataSourceItems.map { items in
+        items.compactMap { item -> Reward? in item.rewardAddOnCardViewData?.reward }
+      }
+    )
+
     let totalSelectedAddOnsQuantity = Signal.combineLatest(
       latestSelectedQuantities,
-      baseReward
+      baseReward.map(\.id),
+      allRewards.map { $0.map(\.id) }
     )
-    .map { quantities, baseReward in
+    .map { quantities, baseRewardId, addOnRewardIds in
       quantities
         // Filter out the base reward for determining the add-on quantities
-        .filter { key, _ in key != baseReward.id }
+        .filter { key, _ in key != baseRewardId && addOnRewardIds.contains(key) }
```

Comment:
This change ensures that if an add-on selection is made, and the shipping location is changed to one that no longer includes that add-on, that it will not be included in the total quantity displayed on the CTA button. Changing the filter back restores the previous selection in the UI. Tapping continue will only pass add-ons through to the pledge view that are visible in the current selection.